lib/supavisor/db_handler.ex

```diff
@@ -14,15 +14,20 @@ defmodule Supavisor.DbHandler do
   alias Supavisor.{Monitoring.Telem, Protocol.Server}
 
   @reconnect_timeout 2_500
+  @sock_closed [:tcp_closed, :ssl_closed]
 
   def start_link(config) do
     :gen_statem.start_link(__MODULE__, config, hibernate_after: 5_000)
   end
 
   @spec call(pid(), pid(), binary()) :: :ok | {:error, any()} | {:buffering, non_neg_integer()}
-  def call(pid, caller, msg) do
-    :gen_statem.call(pid, {:db_call, caller, msg}, 15_000)
-  end
+  def call(pid, caller, msg), do: :gen_statem.call(pid, {:db_call, caller, msg}, 15_000)
+
+  @spec get_state(pid()) :: any()
```

Comment:
Should this be a list of possibe states as atoms?

Reply:
yeah, that would be more readable